src/solvers/dg_curved/dg.jl

```diff
@@ -16,6 +16,32 @@ end
 end
 
 
+@inline function calc_boundary_flux_by_direction!(surface_flux_values::AbstractArray{<:Any,4}, u, t, orientation,
```

Comment:
If this is to work for 3D, you might need to remove the `::AbstractArray` since in 3D it's probably `ndims = 5` for the array, isn't it?